src/main/java/ch/njol/skript/events/EvtPlayerArmorChange.java

```diff
@@ -0,0 +1,91 @@
+package ch.njol.skript.events;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.lang.Literal;
+import ch.njol.skript.lang.SkriptEvent;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.SyntaxStringBuilder;
+import ch.njol.skript.registrations.Classes;
+import com.destroystokyo.paper.event.player.PlayerArmorChangeEvent;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.EquipmentSlot;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.concurrent.CompletionException;
+
+public class EvtPlayerArmorChange extends SkriptEvent {
+
+	private static boolean BODY_SLOT_EXISTS;
+
+	static {
+		if (Skript.classExists("com.destroystokyo.paper.event.player.PlayerArmorChangeEvent")) {
+			Skript.registerEvent("Armor Change", EvtPlayerArmorChange.class, PlayerArmorChangeEvent.class,
+				"[player] armo[u]r change[d] [(from|at|of) %-equipmentslots%]",
+					"[player] %equipmentslot% change[d]")
+				.description("Called when armor pieces of a player are changed.")
+				.requiredPlugins("Paper")
+				.keywords("armor", "armour")
+				.examples(
+					"on armor change:",
+						"\tbroadcast the old armor item",
+					"on player armour changed from helmet slot:",
```

Comment:
i think this format should be removed entirely
it's very clunky and circuitous and the `on helmet change` style solves the problem much more nicely